ister_gui.py

```diff
@@ -641,24 +643,215 @@ def validate_condition(self):
+            ipaddress.ip_address(self.static_ip.get_edit_text())
+        except Exception as err:
+            self.error = 'The configuration for "ip" is invalid {}'.format(err)
+            # an empty action signals to refresh the page
+            self._action = ''
+        if self.interface.get_edit_text() not in self.ifaceaddrs.keys():
+            self.error = 'Interface "{}" not detected'\
+                         .format(self.interface.get_edit_text())
+            # an empty action signals to refresh the page
+            self._action = ''
+
+        if not self.error:
+            config = self.config
+            return self._action
+        else:
+            Alert('Error!', self.error).do_alert()
+            self.error = ''
+            return self._action
+
+    def build_ui_widgets(self):
+        """Build ui handler
+        The last urwid Divider helps the tab movement to work"""
+        fmt = '{0:>25}'
+        wired = '* Connection to clearlinux.org: '
+        wired += 'established' if self._network_connection() else \
+                 'none detected, install will fail'
+        wired_req = urwid.Text(wired)
+        interface_ip = self._find_interface_ip()
+        interface_msg = 'Interface: '
+        ip_msg = 'IP address: '
+        if interface_ip:
+            interface_res = interface_ip[0]
+            ip_res = interface_ip[1]
+        else:
+            interface_res = 'none found'
+            ip_res = 'none found'
+
+        self.interface = urwid.Edit(fmt.format(interface_msg),
+                                    interface_res)
+        self.static_ip = urwid.Edit(fmt.format(ip_msg), ip_res)
+        # pylint: disable=E1103
+        try:
+            gateway = netifaces.gateways()['default'][netifaces.AF_INET][0]
+        except Exception:
+            gateway = 'none found'
+        # pylint: enable=E1103
+
+        self.gateway = urwid.Edit(fmt.format('Gateway: '), gateway)
+        self._ui_widgets = [wired_req,
+                            urwid.Divider(),
+                            self.installer_proxy,
+                            urwid.Divider(),
+                            self.proxy_col,
+                            urwid.Divider(),
+                            self.interface,
+                            self.static_ip,
+                            self.gateway,
+                            urwid.Divider(),
+                            self.static_col,
+                            urwid.Divider(),
+                            self.reset_col,
+                            urwid.Divider()]
+
+    def build_ui(self):
+        self._ui = SimpleForm(u'Network Requirements',
+                              self._ui_widgets, buttons=["Refresh", "Ok"])
+
+    def run_ui(self):
+        return self._ui.do_form()
+
+    def _network_connection(self):
+        """Check if connection to https://www.clearlinux.org is available"""
+        cmd = ['/usr/bin/curl', '-s', '-I', 'https://www.clearlinux.org',
```

Comment:
The installation images do not have curl by default, see comment in ister.py on validating network that kind of goes with this. Seems the desire was to use pycurl.